tooling/miniutil/src/build/ty.rs

```diff
@@ -12,10 +12,28 @@ pub fn ref_ty(pointee: PointeeInfo) -> Type {
     Type::Ptr(PtrType::Ref { mutbl: Mutability::Immutable, pointee })
 }
 
+pub fn ref_ty_for(ty: Type) -> Type {
+    ref_ty(PointeeInfo {
+        layout: ty.layout::<x86_64>(),
+        inhabited: true,
+        freeze: true,
+        unpin: true,
+    })
+}
+
 pub fn ref_mut_ty(pointee: PointeeInfo) -> Type {
     Type::Ptr(PtrType::Ref { mutbl: Mutability::Mutable, pointee })
 }
 
+pub fn ref_mut_ty_for(ty: Type) -> Type {
+    ref_mut_ty(PointeeInfo {
+        layout: ty.layout::<x86_64>(),
+        inhabited: true,
+        freeze: true,
+        unpin: true,
```

Comment:
The fact that you hard-code these defaults makes this function quite confusing. It's not actually correct for all types.

Reply:
These flags say, e.g., whether the type is `Freeze` which indicates that there is no `UnsafeCell` in this type. Minimize computes this from Rust types, but for minitest we need to set this ourselves.

I think this is fine for test use but the name should indicate that. Maybe `ref_mut_ty_default_markers_for`? And then the doc comment should explain the rest. (`Freeze` and `Unpin` are marker traits, hence "default markers". I am open to other names.)

Reply:
Okay. Are these flags given for a trait object?